server/src/static/js/parent-helper.js

```diff
@@ -2,17 +2,29 @@
 /*global console */
 
 let loaded = false,
-  height = null;
+  height = null,
+  resolveChildReference;
+
+window.childReferencePromise = new Promise((resolve, reject) => {
+  resolveChildReference = resolve;
+});
 
 function doResize() {
   document.getElementById("frame").height = height;
 }
 
 window.onmessage = function(m) {
-  if (m.origin !== location.origin) {
-    console.warn("Parent iframe received message from unexpected origin:", m.origin, "instead of", location.origin);
+  if (m.origin !== POST_MESSAGE_ORIGIN) {
+    console.warn("Parent iframe received message from unexpected origin:", m.origin, "instead of", POST_MESSAGE_ORIGIN);
     return;
   }
+  if (resolveChildReference) {
```

Comment:
I don't think it's any more code, just a clearer function that encapsulates a bit more of the complexity of postMessage